src/c.ml

```diff
@@ -124,3 +131,11 @@ module Sources = struct
     in
     {Kind.Dict. c; cxx}
 end
+
+let all_possible_exts =
+  let exts = Kind.possible_exts ~dune_version:Stanza.latest_version in
+  header_ext :: exts C @ exts Cxx
```

Comment:
FWIW, many people call their headers something other than ".h". It seems weird that we're getting away with just ".h" here.

Reply:
That's true, but we can't fix everything at once. Let's wait for someone to at least report this issue.

Reply:
Indeed.